core/src/main/java/org/lfenergy/shapeshifter/core/model/UftpMessage.java

```diff
@@ -30,8 +30,11 @@ public static <T extends PayloadMessageType> UftpMessage<T> createOutgoing(UftpP
    * @param <U> The type of message that is being referenced (e.g. FlexRequest).
    * @return Reference to the referenced message.
    */
-  public <U extends PayloadMessageType> UftpMessageReference<U> referenceToPreviousMessage(String referencedMessageID, Class<U> referencedType) {
+  public <U extends PayloadMessageType> UftpMessageReference<U> findReferenceMessageInConversation(String referencedMessageID,
```

Comment:
The verb "find" implies (at least to me) that this method will actually start finding it, but actually it only creates a UftpMessageReference value object to hold the relevant reference values. So I think this method should be renamed to something no implying that it starts finding something.

Reply:
`findMessageReferenceInConversation`?

Reply:
i've reverted it to `referenceToPreviousMessage`